src/dso_api/dynamic_api/middleware.py

```diff
@@ -27,28 +27,29 @@ def process_view(self, request, view_func, view_args, view_kwargs):
         return None
 
 
-class TemporalDatasetMiddleware(MiddlewareMixin):
+class TemporalTableMiddleware(MiddlewareMixin):
     """
-    Assign `versioned`, `dateset_verison` and `temporal_slice` to request.
+    Assign `versioned`, `table_version` and `temporal_slice` to request.
     """
 
     def process_request(self, request):
         request.versioned = False
-        request.dataset_version = None
-        request.dataset_temporal_slice = None
+        request.table_version = None
+        request.table_temporal_slice = None
 
     def process_view(self, request, view_func, view_args, view_kwargs):
-        if not hasattr(request, "dataset") or request.dataset.temporal is None:
+        if not hasattr(request, "dataset") or not view_func.cls.model.is_temporal():
```

Comment:
Dit gaat crashen op niet class-based views. `view_func.cls` kan een attributeerror geven, de view heeft wellicht geen model. Ging dit niet al mis bij de unit tests / OpenAPI?

Reply:
Het gaat niet mis omdat view_func.cls moet bestaan als request.dataset bestaat. Maar dit expliciet checken met een try/except is beter idd.

Reply:
ah check, vandaar!